go/exocom/test_helpers/mock_service.go

```diff
@@ -69,10 +71,9 @@ func (m *MockService) Close() error {
 
 //GetReceivedMessages gets the messages this mock service has received
 func (m *MockService) GetReceivedMessages() []structs.Message {
-	if m.ReceivedMessages == nil {
-		return []structs.Message{}
-	}
-	return m.ReceivedMessages
+	m.receivedMessagesMutex.RLock()
+	defer m.receivedMessagesMutex.RUnlock()
+	return m.receivedMessages
```

Comment:
Just out of curiosity: is the act of simply returning a variable already a race condition?